Adjust the Help Panel to be used anywhere.

afercia opened the pull request:
## Summary

This PR can be summarized in the following changelog entry:

* Update the Admin Help Panel classo to be used in the settings pages too

## Relevant technical choices:

- moved the CSS to admin global scss
- changed the button color globally so it will be purple also in the metabox and in the search console
- kept the class untouched, but it could probably benefit from a small refactoring to pass an $args array to be used, for example, for the optional wrapper and additional CSS classes; I'd like to review this part later if no objections
- considered a small helper function to print out the help button and content but then decided to not do that as the current class allows to print out them separately, and that's nice
- the styling could be improved, open to suggestions

## Test instructions

This PR can be tested by following these steps:

- build the CSS
- see usage example in `admin/views/tabs/metas/general.php`, the class needs:
  - an id
  - the visually hidden button text
  - the help text

Fixes #8886

Screenshots:

![screen shot 2018-02-09 at 15 14 18](https://user-images.githubusercontent.com/1682452/36032238-a8755880-0dad-11e8-98bc-403e2885f154.png)

![screen shot 2018-02-09 at 15 14 24](https://user-images.githubusercontent.com/1682452/36032239-a88e5c04-0dad-11e8-92d1-8e7000d21330.png)

afercia commented:
Note: the new span with `aria-hidden="true"` is a small improvement to completely hide the icon from assistive technologies.

jcomack commented:
CR done 👍

jcomack commented:
Acceptance failed 🚧 

There are few things that need some attention:

1) In https://github.com/Yoast/wordpress-seo/issues/8721 we'll be introducing a new color for the question marks. As a preparation, it'd be nice if the gray color that has been decided on, is already implemented in this PR (Discussed this with @omarreiss)
2) There's only a single toggle being introduced in this PR, despite there being more descriptions that could use this toggle feature. For example, the Search Appearance page is missing toggles for:

    - Title Separator
    - Media & attachment URLs
    - Category URLs
    - Archives (has a long description block at the top)
    - Special Pages
    - How to insert breadcrumbs in your theme (might be a good candidate, but I'm not entirely sure about that one)
    - RSS

3) The Social page has multiple tabs that could also benefit from the toggle being implemented.

4) The text feels misaligned with the rest of the elements on the page (see screenshot)
![search_appearance_-_yoast_seo_ _acceptance_test_ _wordpress](https://user-images.githubusercontent.com/4181340/36150353-b45993be-10c3-11e8-8ddb-d82554e7ccb9.png)

afercia commented:
About 2) and 3) please notice on trunk some settings pages still need to be updated and re-arranged. I guess that needs to happen before using the new Help. As noted in the test instructions, this PR contains just an "usage example".
Worth noting:
- Category URLs: that text is actually the label of the control and can't be hidden
- Archives (has a long description block at the top): there's no title where to place the Help button; the description refers to both the author archives and the date archives, so it can't be placed under one of those
- some of the descriptions in the Social page are actually descriptions of the "switch toggles" and will be addressed in #8804 for which this PR is a blocker

> 4. The text feels misaligned with the rest of the elements on the page

The current max-width of 30 em is the one originally used in the metabox. As noted in the PR description, the styling could be improved but that needs design feedback /cc @hedgefield

hedgefield commented:
The width thing is tricky because sticking to our own max-word-per-line rule would be preferred, but that is probably the cause of it looking weird now. I'd say adjust it to at least stretch out to the end of the input fields/second column, it's not _that_ much wider.

This should be applied on the Features page too, there the descriptions had no max width.

afercia commented:
In the Search Appearance - General tab, what to do when in "Reading Settings" the front page is set to static page and this appears? (see below)

<img width="710" alt="screen shot 2018-02-13 at 15 47 43" src="https://user-images.githubusercontent.com/1682452/36156624-f12a315e-10d7-11e8-8574-4f0026d0f7f5.png">

Maybe just keep both paragraphs as they're, not in italic? Open to suggestions.

afercia commented:
Added the Help Panel in various places, where it made sense to me. Please let me know if that's OK and if there are other places that could benefit from the same treatment.

About the archives settings, I've added a paragraph before the Author / Date archives settings because the help text refers to both settings. Open to better ideas:

<img width="728" alt="screen shot 2018-02-14 at 09 57 30" src="https://user-images.githubusercontent.com/1682452/36196005-9fbb0148-116f-11e8-8f1a-f09d93b3107c.png">

afercia commented:
To test, remember to build the CSS.

jcomack commented:
Acceptance 🚧 

There are a few more sections that I feel could benefit from these toggles.

Social -> Facebook -> Add Open Graph meta data
Social -> Facebook -> Facebook Insights and Admins
Social -> Twitter -> Add Twitter card meta data
Social -> Pinterest (Although you'll run into that (?) beneath the `Need Help?` button situation again)
Social -> Google+ (Same story as for Pinterest)


I'm not sure how far this would cover all the situations @jdevalk had envisioned when coming up with this idea, so perhaps it's a good idea if he gives his two cents on this matter as well.

afercia commented:
@jcomack @jdevalk this PR is a blocker for 2 other changes, see #8804 and #8926. I'd suggest, if possible, to move this PR on and split the remaining possible improvements in a separate issue?

@jcomack some of the places you've suggested are actually output by the "switch toggles" and the Help Panel should be implemented within their functions, see #8804

jcomack commented:
Okay, what I'm going to do it re-test and merge this PR so the other issues can be tackled. I'll additionally make a separate issue / checklist for all the places that could benefit from the question mark.

jcomack commented:
Acceptance done 👍
